src/QMCWaveFunctions/EinsplineSetBuilderESHDF.fft.cpp

```diff
@@ -124,106 +126,121 @@ bool EinsplineSetBuilder::ReadOrbitalInfo_ESHDF(bool skipChecks)
   // Read atom orbital info from xml //
   /////////////////////////////////////
   // construct Super2Prim mapping.
-  if (Super2Prim.size() == 0)
-  {
-    //SourcePtcl->convert2Cart(SourcePtcl->R);
-    Super2Prim.resize(SourcePtcl->R.size(), -1);
-    std::vector<int> prim_atom_counts;
-    prim_atom_counts.resize(IonPos.size(), 0);
-    for (int i = 0; i < SourcePtcl->R.size(); i++)
+  // Check the number of ions in the XML and 
+  // HDF5. Skip the checks in some situations but 
+  // not others. See Issue #2224.
+
+  if (SourcePtcl->R.size() != IonPos.size()) {
+    app_error() << "The ion sizes differ between the XML and HDF5.\n" 
+                << "Number of ions in XML : " << SourcePtcl->R.size() << std::endl
+                << "Number of ions in HDF5: " << IonPos.size() << std::endl;
+    abort();
+  }
+
+  if (SourcePtcl->R.size() == 0 && IonPos.size() == 0) {
+    app_log() << "There are no ions in XML nor HDF5. No further checks will be performed.\n";
+  } else {
+    if (Super2Prim.size() == 0)
```

Comment:
You might also use the previous check to set a variable (checkCellConsistency)
```
bool checkCellConsistency = true;
if (SourcePtcl ....)
   // message
   checkCellConsistency = false;
}
if (Super2Prim.size() == 0 && checkCellConsistency)
{
...
```

That would avoid indenting a bunch of code.   Unfortunately, the diff algorithm is not recognizing this block of code is the same as before with added indentation, and it makes the changes look more extensive than they really are.

Reply:
yeah, I agree. This is what I will do.